arangopipe/tests/CItests/arangopipe_testcases.py

```diff
@@ -0,0 +1,423 @@
+    
+    def test_register_model(self):
+        err_raised = False
+        try:
+            self.register_model()
+        except:
+            err_raised = True
+            print ('-'*60)
+            traceback.print_exc(file=sys.stdout)
+            print ('-'*60)
+            self.assertTrue(err_raised,\
+                            'Exception raised while registering model')
+        self.assertFalse(err_raised)
+        return
+    
+    def test_lookup_model(self):
+        err_raised = False
+        try:
+            self.register_model()
+            self.lookup_model()
+        except:
+            err_raised = True
+            print ('-'*60)
+            traceback.print_exc(file=sys.stdout)
+            print ('-'*60)
+            self.assertTrue(err_raised,\
+                            'Exception raised while looking up model')
+        self.assertFalse(err_raised)
+        return
+    
+    def test_log_run(self):
+        err_raised = False
+        try:
+            self.register_dataset()
+            self.register_featureset()
+            self.register_model()
+            self.log_run()
+        except:
+            err_raised = True
+            print ('-'*60)
+            traceback.print_exc(file=sys.stdout)
+            print ('-'*60)
+            self.assertTrue(err_raised,\
+                            'Exception raised while logging performance')
+        self.assertFalse(err_raised)
+        return
+    
+    def test_provision_deployment(self):
+        err_raised = False
+        try:
+            self.register_dataset()
+            self.register_featureset()
+            self.register_model()
+            self.log_run()
+            self.provision_deployment()
+        
+        except:
+            err_raised = True
+            print ('-'*60)
+            traceback.print_exc(file=sys.stdout)
+            print ('-'*60)
+            self.assertTrue(err_raised,\
+                            'Exception raised while provisioning deployment')
+        self.assertFalse(err_raised)
+        return
+    
+    def test_log_serving_performance(self):
+        err_raised = False
+        try:
+            self.register_dataset()
+            self.register_featureset()
+            self.register_model()
+            self.log_run()
+            self.provision_deployment()
+            self.log_servingperf()
+        
+        except:
+            err_raised = True
+            print ('-'*60)
+            traceback.print_exc(file=sys.stdout)
+            print ('-'*60)
+            self.assertTrue(err_raised,\
+                            'Exception raised while logging serving performance')
+        self.assertFalse(err_raised)
+        return
+    
+    def test_dataset_shift_positive(self):
+        
+        score = self.dataset_shift_positive()
+        
+        self.assertTrue(score > 0.8)
+        return
+    
+    def test_dataset_shift_negative(self):
+
+        score = self.dataset_shift_negative()
+        
```

Comment:
remove blank line. Would it maybe make sense to use something like https://github.com/google/yapf?

Reply:
I will look at a formatting tool. I'll open an issue for formatting.